### PR TITLE
fix: avoid same name nodes in detection module

### PR DESCRIPTION
**Pull request by tkimura4**
Signed-off-by: tomoya.kimura <tomoya.kimura@tier4.jp>

## Description
avoid to launch duplicated-named nodes in detection module.
```
/perception/object_recognition/detection/clustering/detected_object_feature_remover_($ anon)
/perception/object_recognition/detection/clustering/shape_estimation 
/perception/object_recognition/detection/object_association_merger_($ anon)
```

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by shmpwk — APPROVED** [review]

**Comment by shmpwk** [comment]
Sorry not approved yet

File: launch/tier4_perception_launch/launch/object_recognition/detection/camera_lidar_fusion_based_detection.launch.xml [hunk 1]
```diff
@@ -64,11 +64,13 @@
     <include file="$(find-pkg-share shape_estimation)/launch/shape_estimation.launch.xml">
       <arg name="input/objects" value="clusters"/>
       <arg name="output/objects" value="objects_with_feature"/>
+      <arg name="node_name" value="shape_estimation_clustering" />
     </include>
     <!-- convert DynamicObjectsWithFeatureArray to DynamicObjects -->
     <include file="$(find-pkg-share detected_object_feature_remover)/launch/detected_object_feature_remover.launch.xml">
       <arg name="input" value="objects_with_feature"/>
       <arg name="output" value="objects"/>
+    <arg name="node_name" value="detected_object_feature_remover_clustering" />
```
Comment (shmpwk) [inline launch/tier4_perception_launch/launch/object_recognition/detection/camera_lidar_fusion_based_detection.launch.xml:73]:
```suggestion
      <arg name="node_name" value="detected_object_feature_remover_clustering" />
```

File: launch/tier4_perception_launch/launch/object_recognition/detection/camera_lidar_fusion_based_detection.launch.xml [hunk 2]
```diff
@@ -103,11 +105,13 @@
     <include file="$(find-pkg-share shape_estimation)/launch/shape_estimation.launch.xml">
       <arg name="input/objects" value="camera_lidar_fusion/clusters"/>
       <arg name="output/objects" value="camera_lidar_fusion/objects_with_feature"/>
+      <arg name="node_name" value="shape_estimation_camera_lidar_fusion" />
     </include>
     <!-- convert DynamicObjectsWithFeatureArray to DynamicObjects -->
     <include file="$(find-pkg-share detected_object_feature_remover)/launch/detected_object_feature_remover.launch.xml">
       <arg name="input" value="camera_lidar_fusion/objects_with_feature"/>
       <arg name="output" value="camera_lidar_fusion/objects"/>
+    <arg name="node_name" value="detected_object_feature_camera_lidar_fusion" />
```
Comment (shmpwk) [inline launch/tier4_perception_launch/launch/object_recognition/detection/camera_lidar_fusion_based_detection.launch.xml:114]:
```suggestion
      <arg name="node_name" value="detected_object_feature_camera_lidar_fusion" />
```

File: launch/tier4_perception_launch/launch/object_recognition/detection/lidar_based_detection.launch.xml [hunk 3]
```diff
@@ -96,11 +96,13 @@
       <arg name="input/object0" value="$(var merger/input/objects)"/>
       <arg name="input/object1" value="clustering/objects"/>
       <arg name="output/object" value="temporary_merged_objects"/>
+      <arg name="node_name" value="object_association_merger_unknown_clustering" />
     </include>
     <include file="$(find-pkg-share object_merger)/launch/object_association_merger.launch.xml">
       <arg name="input/object0" value="temporary_merged_objects"/>
       <arg name="input/object1" value="detection_by_tracker/objects"/>
       <arg name="output/object" value="objects_before_filter"/>
+    <arg name="node_name" value="object_association_merger_detection_by_tracker" />
```
Comment (shmpwk) [inline launch/tier4_perception_launch/launch/object_recognition/detection/lidar_based_detection.launch.xml:105]:
```suggestion
      <arg name="node_name" value="object_association_merger_detection_by_tracker" />
```

**Review by shmpwk — COMMENTED** [review]
It works fine

**Review by shmpwk — APPROVED** [review]

File: perception/detected_object_feature_remover/launch/detected_object_feature_remover.launch.xml [hunk 4]
```diff
@@ -1,8 +1,9 @@
 <launch>
   <arg name="input"/>
   <arg name="output"/>
+  <arg name="node_name" default="detected_object_feature_remover"/>
 
-  <node pkg="detected_object_feature_remover" exec="detected_object_feature_remover" name="$(anon detected_object_feature_remover)" output="screen">
+  <node pkg="detected_object_feature_remover" exec="detected_object_feature_remover" name="$(var node_name)" output="screen">
```
Comment (yukkysaito) [inline perception/detected_object_feature_remover/launch/detected_object_feature_remover.launch.xml:6]: @tkimura4 why is it changed?

**Pull request merged**